frontend/src/features/public-form/components/StatusTrackerPage/StatusTrackerPage.tsx

```diff
@@ -116,6 +117,10 @@ export const StatusTrackerPage = (): JSX.Element => {
           : WorkflowStatus.COMPLETED
         : WorkflowStatus.PENDING
```

Comment:
Can this be extracted to a separate function to make this more readable? 


```suggestion
const getWorkflowStatus = () => {
  // Step hasn't been submitted yet
  if (index >= submittedSteps.length) {
    return WorkflowStatus.PENDING
  }

  // Step has been submitted
  const currentWorkflowStep = workflow[index]
  
  // If it's not an approval step, it's completed
  if (!currentWorkflowStep.approval_field) {
    return WorkflowStatus.COMPLETED
  }

  // It's an approval step - check if status exists in submitted step
  if ('status' in submittedStep) {
    return submittedStep.status
  }

  // Approval step without status defaults to completed
  return WorkflowStatus.COMPLETED
}
```

Reply:
I've refactored this out into a `getWorkflowStatusFromFormResponse` function and called it here.